src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -123,9 +124,12 @@ const enum VSCodeOscPt {
 export class ShellIntegrationAddon extends Disposable implements IShellIntegration, ITerminalAddon {
 	private _terminal?: Terminal;
 	readonly capabilities = new TerminalCapabilityStore();
+	private _hasUpdatedTelemetry: boolean = false;
+	private _activationTimeout!: any;
 
 	constructor(
-		@ILogService private readonly _logService: ILogService
+		@ILogService private readonly _logService: ILogService,
+		private readonly _telemetryService?: ITelemetryService
```

Comment:
This needs to go first for dependency injection to work for `ILogService`

Reply:
it can't because it's optional.

Reply:
figured it out